aesara/link/utils.py

```diff
@@ -588,6 +606,11 @@ def compile_function_src(src, function_name, global_env=None, local_env=None):
     if local_env is None:
         local_env = {}
 
+    global_env["use_optimized_cheap_pass"] = use_optimized_cheap_pass
+    src = f"""
+with use_optimized_cheap_pass():
+{indent(src, " " * 4)}
+"""
```

Comment:
Same here; this function is supposed to be applicable to _any_ string-to-Python-code conversion, so we can't add Numba-specific code to it.